### PR TITLE
[Runtime] Avoid +class overrides when initializing an ObjC class.

### PR DESCRIPTION
**Pull request by mikeash**
swift_getInitializedObjCClass called [c class] to trigger class initialization, and returned the value. This wreaked havoc when the class in question overrides +class. Instead, ignore the return value and return c. Switch from +class to +self, which is much less likely to be overridden. Calling an overridden method could have performance downsides or even cause unwanted side effects.

rdar://problem/49853091

### CODE REVIEW

**Comment by mikeash** [comment]
@swift-ci please test

**Comment by jrose-apple** [comment]
I think it's possible to write a terrible test case for this that would fail the old way.

**Comment by mikeash** [comment]
I think you're right! `+class { return nil; }` ought to do it, along with something that triggers this path. I'll give that a try.

**Comment by mikeash** [comment]
That does indeed trigger a failure with the old code. New code works.

**Comment by mikeash** [comment]
@swift-ci please test

File: test/stdlib/ObjCEvilClassInitialization.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// RUN: mkdir -p %t
```
Comment (jrose-apple) [inline test/stdlib/ObjCEvilClassInitialization.swift]: Nitpick: `%empty-directory(%t)` will also clear out any contents left over from the last run.

**Review by jrose-apple — APPROVED** [review]

File: test/stdlib/ObjCEvilClassInitialization.swift (hunk 1 above)
Comment (mikeash) [reply test/stdlib/ObjCEvilClassInitialization.swift]: Thanks. We have a bunch of tests with each one, and I happened to crib from one with `mkdir`. Seems better to do it with `empty-directory` though.

**Comment by mikeash** [comment]
@swift-ci please test

**Pull request merged**